src/tests/test_typeconversions.py

```diff
@@ -26,6 +26,182 @@ def test_string_conversion(selenium_module_scope, s):
+        strategies.integers(min_value=-(2 ** 53), max_value=2 ** 53),
+        strategies.floats(allow_nan=False),
+    )
+)
+@settings(deadline=600)
+def test_number_conversions(selenium_module_scope, n):
+    with selenium_context_manager(selenium_module_scope) as selenium:
+        import json
+
+        s = json.dumps(n)
+        selenium.run_js(
+            f"""
+            window.x_js = eval({s!r}); // JSON.parse apparently doesn't work
+            pyodide.runPython(`
+                import json
+                x_py = json.loads({s!r})
+            `);
+            """
+        )
+        assert selenium.run_js(f"""return pyodide.runPython('x_py') === x_js;""")
+        assert selenium.run(
+            """
+            from js import x_js
+            x_js == x_py
+            """
+        )
+
+
+def test_nan_conversions(selenium):
+    selenium.run_js(
+        """
+        window.a = NaN;
+        pyodide.runPython(`
+            from js import a
+            from math import isnan, nan
+            assert isnan(a)
+        `);
+        let b = pyodide.runPython("nan");
+        if(!Number.isNaN(b)){
+            throw new Error();
+        }
+        """
+    )
+
+
+@given(n=strategies.integers())
+@settings(deadline=600)
+def test_bigint_conversions(selenium_module_scope, n):
+    with selenium_context_manager(selenium_module_scope) as selenium:
+        h = hex(n)
+        selenium.run_js(
+            """
+            window.assert = function assert(cb){
+                if(cb() !== true){
+                    throw new Error(`Assertion failed: ${cb.toString().slice(6)}`);
+                }
+            };
+            """
+        )
+        selenium.run_js(f"window.h = {h!r};")
+        selenium.run_js(
+            """
+            let negative = false;
+            let h2 = h;
+            if(h2.startsWith('-')){
+                h2 = h2.slice(1);
+                negative = true;
+            }
+            window.n = BigInt(h2);
+            if(negative){
+                window.n = -n;
+            }
+            pyodide.runPython(`
+                from js import n, h
+                n2 = int(h, 16)
+                assert n == n2
+            `);
+            let n2 = pyodide.globals.get("n2");
+            let n3 = Number(n2);
+            if(Number.isSafeInteger(n3)){
+                assert(() => typeof n2 === "number");
+                assert(() => n2 === Number(n));
+            } else {
+                assert(() => typeof n2 === "bigint");
+                assert(() => n2 === n);
+            }
+            """
+        )
+
+
+# Generate an object of any type
+@given(obj=from_type(type).flatmap(from_type))
+@settings(deadline=600)
+def test_hyp_py2js2py(selenium_module_scope, obj):
+    with selenium_context_manager(selenium_module_scope) as selenium:
+        import pickle
+
+        try:
+            assume(obj == obj)  # make sure no NaNs.
+        except:
+            assume(False)
```

Comment:
That's not equivalent to
```
assume(not isnan(x))
```
from the example in https://hypothesis.readthedocs.io/en/latest/details.html#hypothesis.assume, or I guess it fails on arbitrary objects ?

Reply:
I think I tried `assume(not isnan(x))` first and it didn't work, but I can't remember. Equality with `nan` is really weird:
```py
from math import nan
a = [nan]
b = [float('nan')]
a == a # true
b == b # true
a == [nan] # true
a == b # false
b == [nan] # false
b == [float('nan')] # false
```

Reply:
Oh well the current version also supports the case where `x == x` raises an error, which `isnan` I think won't handle.

Reply:
In summary there are three options for `x==x`:
1. True
2. False
3. raises an exception
In cases 2 & 3, we don't want to run the test.

Reply:
Sounds good!